diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter6/CustomGetterSetterRule.kt

```diff
@@ -33,6 +35,21 @@ class CustomGetterSetterRule(configRules: List<RulesConfig>) : DiktatRule(
         val isPrivateSetter = node.getFirstChildWithType(MODIFIER_LIST)?.hasAnyChildOfTypes(PRIVATE_KEYWORD) ?: false
         val isOverrideGetter = node.treeParent.getFirstChildWithType(MODIFIER_LIST)?.hasAnyChildOfTypes(OVERRIDE_KEYWORD) ?: false
 
+        // find matching node
+        val propertyList = node.treeParent?.treeParent?.getAllChildrenWithType(PROPERTY)
```

Comment:
In the new code we suggest to use PSI instead of AstNode, because there is a probability of elimination of this API in future. You can do the following:
```
        val propertyList = node.treeParent
?.treeParent
?.getAllChildrenWithType(PROPERTY)
?.map { it.psi as KtProperty }
```

And now you can instead of `val matchingNode = propertyNode.getFirstChildWithType(IDENTIFIER)` do:

```
                    val matchingNode = propertyNode.name

```

So JetBrains already provides in PSI you all the logic you are implementing in AstNode

Reply:
Moreover with a PSI you can also change `treeParent?.treeParent` logic to something more good-looking.

Reply:
I will change it in next pull-request